src/Symfony/Component/PropertyAccess/AccessInfoGuesser.php

```diff
@@ -0,0 +1,259 @@
+<?php
+
+namespace Symfony\Component\PropertyAccess;
+
+use Psr\Cache\CacheItemPoolInterface;
+use Symfony\Component\Cache\Adapter\NullAdapter;
+use Symfony\Component\Inflector\Inflector;
+
+final class AccessInfoGuesser
```

Comment:
Should have an associated interface if marked final. And I agree that it would fit in PropertyInfo.

Reply:
The return value of `getReadAccessInfo` is really hard to use. You have many cases to handle to get things working the right way. Moreover, it would be hard to apply BC promise on such return type.